File: Task 03/primenumbers.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter the value of n: ', (n) => {
  n = parseInt(n);

  if (isNaN(n) || n <= 0) {
    console.log('Error: n must be a positive integer');
    rl.close();
    return;
  }

  console.log(`\nPrime numbers up to ${n}:`);

  if (n >= 2) {
    console.log(2);
  }

  for (let num = 3; num <= n; num += 2) {
    let isPrime = true;
    for (let i = 3; i * i <= num; i += 2) {
      if (num % i === 0) {
        isPrime = false;
        break;
      }
    }
    if (isPrime) {
      console.log(num);
    }
  }

  rl.close();
});

File: Task 03/primenumbers.rs
use std::io;

fn main() {
    let mut input = String::new();

    println!("Enter the value of n: ");
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");

    let n: u32 = match input.trim().parse() {
        Ok(num) if num > 0 => num,
        _ => {
            println!("Error: n must be a positive integer");
            return;
        }
    };

    println!("\nPrime numbers up to {}:", n);

    if n >= 2 {
        println!("2");
    }

    for num in (3..=n).step_by(2) {
        let mut is_prime = true;
        for i in (3..=(num as f64).sqrt() as u32 + 1).step_by(2) {
            if num % i == 0 {
                is_prime = false;
                break;
            }
        }
        if is_prime {
            println!("{}", num);
        }
    }
}

